src/ui/popup.js

```diff
@@ -572,8 +572,12 @@ export default class Popup extends Evented {
         }
 
         const offsetedPos = pos.add(offset[anchor]).round();
-        DOM.setTransform(this._container, `${anchorTranslate[anchor]} translate(${offsetedPos.x}px,${offsetedPos.y}px)`);
-        applyAnchorClass(this._container, anchor, 'popup');
+        const container = this._container;
+        const _anchor = anchor;
```

Comment:
Nit: given that the closure captures the `anchor` variable and it's not modified later in the method, and that the arrow function retains the context of `this`, why do we need to define aliases here?

Reply:
Oops, I edited the comment here but the old version got posted somehow — so, now I see that the DOM elements are aliased so that it doesn't break in case a task is run after the object is removed from the map, but it's not obvious, so might need additional comments to avoid maintainers tripping up on this. Alternatively, we could look into whether this can be refactored to avoid the issue differently — e.g. perhaps marking a task as cancelled when a corresponding object is removed, or maybe adding some guards for the elements inside the closure.

Reply:
To be honest, I got an error with `flow` regarding `anchor` about it being `null` in the arrow function and I didn't understand how that can be possible, hence the alias.

It would be better however to see how it can be refactored. What about storing the task ID in the popup, as there should always be only one ? Considering the total number of popup, it won't be a burden and then we could cancel the task in `remove()`. Otherwise another simple solution that seems to work is by simply adding `if (this._container && anchor)` in the arrow function.